ui/components/src/ui-components/filtering/FilteringInput.tsx

```diff
@@ -111,27 +157,61 @@ export class FilteringInput extends React.PureComponent<FilteringInputProps, Fil
       return;
 
     this.props.onFilterStart(this.state.searchText);
+    this.setState({ searchStarted: true });
     e.stopPropagation();
   };
 
   private _onInputChanged = (e: React.ChangeEvent<HTMLInputElement>) => {
-    this.setState({ searchText: e.target.value, context: InputContext.ReadyToFilter });
-  };
+    if(this.state.searchStarted){
+      this.props.onFilterCancel();
+    }
+    this.setState({ searchText: e.target.value, searchStarted: false});
+  }
+
 
   /** @internal */
-  public static getDerivedStateFromProps(props: FilteringInputProps, state: FilteringInputState) {
-    if (state.context === InputContext.FilteringInProgress && !props.filteringInProgress) {
-      if (state.searchText && props.resultSelectorProps)
-        return { context: InputContext.FilteringFinished };
-      else
-        return { context: InputContext.FilteringFinishedWithNoStepping };
-    } else if (state.context === InputContext.ReadyToFilter && props.filteringInProgress) {
-      return { context: InputContext.FilteringInProgress };
+  public componentDidUpdate(prevProps: FilteringInputProps, _prevState: FilteringInputState) {
```

Comment:
No need to define the second argument if not used

Reply:
Same for some lambdas in this function

Reply:
Fixed it